internal/format/middle_visitor.go

```diff
@@ -0,0 +1,306 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package format
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/emicklei/proto"
+	"github.com/uber/prototool/internal/settings"
+	"github.com/uber/prototool/internal/text"
+)
+
+type middleVisitor struct {
+	*baseVisitor
+
+	isProto2          bool
+	rpcUseSemicolons  bool
+	haveHitNonComment bool
+	parent            proto.Visitee
+}
+
+func newMiddleVisitor(config settings.Config, isProto2 bool) *middleVisitor {
+	return &middleVisitor{isProto2: isProto2, rpcUseSemicolons: config.Format.RPCUseSemicolons, baseVisitor: newBaseVisitor(config.Format.Indent)}
+}
+
+func (v *middleVisitor) Do() []*text.Failure {
+	return v.Failures
+}
+
+func (v *middleVisitor) VisitMessage(element *proto.Message) {
+	v.haveHitNonComment = true
+	v.PComment(element.Comment)
+	prefix := "message "
+	if element.IsExtend {
+		prefix = "extend "
+	}
+	if len(element.Elements) == 0 {
+		v.P(prefix, element.Name, " {}")
+		v.P()
+		return
+	}
+	v.P(prefix, element.Name, " {")
+	v.In()
+	originalParent := v.parent
+	v.parent = element
+	for _, child := range element.Elements {
+		child.Accept(v)
+	}
+	v.parent = originalParent
+	v.Out()
+	v.P("}")
+	if v.parent == nil {
+		v.P()
+	}
+}
+
+func (v *middleVisitor) VisitService(element *proto.Service) {
+	v.haveHitNonComment = true
+	v.PComment(element.Comment)
+	if len(element.Elements) == 0 {
+		v.P("service ", element.Name, " {}")
+		v.P()
+		return
+	}
+	v.P("service ", element.Name, " {")
+	v.In()
+	originalParent := v.parent
+	v.parent = element
+	for _, child := range element.Elements {
+		child.Accept(v)
+	}
+	v.parent = originalParent
+	v.Out()
+	v.P("}")
+	v.P()
+}
+
+func (v *middleVisitor) VisitSyntax(element *proto.Syntax) {
+	// done in first pass visitor
+	v.haveHitNonComment = true
+}
+
+func (v *middleVisitor) VisitPackage(element *proto.Package) {
+	// done in first pass visitor
+	v.haveHitNonComment = true
+}
+
+func (v *middleVisitor) VisitOption(element *proto.Option) {
+	v.haveHitNonComment = true
+	// file options done in first pass visitor
+	if v.parent == nil {
+		return
+	}
+	switch v.parent.(type) {
+	case (*proto.Enum):
+		v.POptions(false, element)
+	case (*proto.Message):
+		v.POptions(false, element)
+	case (*proto.Oneof):
+		v.POptions(false, element)
+	case (*proto.Service):
+		v.POptions(false, element)
+	default:
+		v.AddFailure(element.Position, "unhandled child option")
+	}
+}
+
+func (v *middleVisitor) VisitImport(element *proto.Import) {
+	// done in first pass visitor
+	v.haveHitNonComment = true
+}
+
+func (v *middleVisitor) VisitNormalField(element *proto.NormalField) {
+	v.haveHitNonComment = true
+	prefix := ""
+	if element.Repeated {
+		prefix = "repeated "
+	}
+	if v.isProto2 {
+		// technically these are only set if the file is proto2
+		// but doing this just to make srue
```

Comment:
nit: srue -> sure

Reply:
Idk how you find all these comment spelling errors, it's impressive